src/Control/HTTP.php

```diff
@@ -323,41 +364,45 @@ public static function get_mime_type($filename)
     /**
      * Set the maximum age of this page in web caches, in seconds.
      *
+     * @deprecated 4.2..5.0 Use HTTPCacheControlMiddleware::singleton()->setMaxAge($age) instead
      * @param int $age
      */
     public static function set_cache_age($age)
     {
+        Deprecation::notice('5.0', 'Use HTTPCacheControlMiddleware::singleton()->setMaxAge($age) instead');
         self::$cache_age = $age;
+        HTTPCacheControlMiddleware::singleton()->setMaxAge($age);
     }
 
     /**
      * @param string $dateString
+     * @deprecated 4.2..5.0 Use HTTPCacheControlMiddleware::registerModificationDate() instead
      */
     public static function register_modification_date($dateString)
     {
-        $timestamp = strtotime($dateString);
-        if ($timestamp > self::$modification_date) {
-            self::$modification_date = $timestamp;
-        }
+        Deprecation::notice('5.0', 'Use HTTPCacheControlMiddleware::registerModificationDate() instead');
+        HTTPCacheControlMiddleware::singleton()->registerModificationDate($dateString);
```

Comment:
Let's delete this functionality altogether, and leave the deprecated methods as a no-op.

Reply:
ChangeDetectionMiddleware

Reply:
+1 on what Sam said. The whole idea around using `DataObject->LastEdited` for `register_modification_date()` is flawed, because it ignores any cached data which doesn't instanciate those objects. I'm aware of this leading to issues in a recent project. And rather than trying to make it more smart (tag caches with this metadata etc), we should remove it altogether. ETag is a better way to address this.